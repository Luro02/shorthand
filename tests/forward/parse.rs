use shorthand::ShortHand;

#[derive(ShortHand)]
struct Example {
    #[shorthand(enable(forward))] // valid
    value_0: String,
    #[shorthand(enable(forward(doc)))] // valid
    value_1: String,
    #[shorthand(disable(forward(doc)))] // valid
    value_2: String,
    #[shorthand(enable(forward("")))] // invalid
    value_3: String,
    #[shorthand(enable(forward = ""))] // invalid
    value_4: String,
    #[shorthand(enable(forward(x = "")))] // invalid
    value_5: String,
    #[shorthand(enable(""))] // invalid
    value_6: String,
    #[shorthand(enable(forward(x(y))))] // invalid
    value_7: String,
}

fn main() {}
