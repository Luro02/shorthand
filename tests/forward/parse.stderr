error: unexpected literal type `string`
  --> $DIR/parse.rs:11:32
   |
11 |     #[shorthand(enable(forward("")))] // invalid
   |                                ^^

error: unexpected meta-item format `NameValue`, expected `Path` or `List`
  --> $DIR/parse.rs:13:24
   |
13 |     #[shorthand(enable(forward = ""))] // invalid
   |                        ^^^^^^^^^^^^

error: unexpected meta-item format `NameValue`, expected `Path`
  --> $DIR/parse.rs:15:32
   |
15 |     #[shorthand(enable(forward(x = "")))] // invalid
   |                                ^^^^^^

error: unexpected literal type `string`
  --> $DIR/parse.rs:17:24
   |
17 |     #[shorthand(enable(""))] // invalid
   |                        ^^

error: unexpected meta-item format `List`, expected `Path`
  --> $DIR/parse.rs:19:32
   |
19 |     #[shorthand(enable(forward(x(y))))] // invalid
   |                                ^^^^
